commit 78d3eaaf7a16fa6941245f7d71234ad22964cd7e (refs/changes/35/5335/2)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2019-01-09T12:14:04-05:00 (9 months ago)
    
    OS-7495 VM create should wait for zfs properties to be present in vminfod
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index a9f863ee..c8af75c2 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -8130,10 +8130,29 @@ function installZone(payload, log, callback)
                 var cancelFn;
                 vasync.parallel({funcs: [
                     function (cb2) {
+                        /*
+                         * Ensure that the following properties are/become
+                         * available on the VM while `zoneadm install` is
+                         * called.
+                         *
+                         * Bhyve zones, for example, will already have the
+                         * zfs/zpool properties available as the call to
+                         * createVM() before this will have created them
+                         * manually.  Other zones will have these properties
+                         * added by the call to `zoneadm install` as the image
+                         * is cloned into place.
+                         *
+                         * Regardless of method, this logic below will wait on
+                         * vminfod for all of the properties to be set to their
+                         * proper values before moving on.
+                         */
                         var obj = {
                             uuid: payload.uuid,
                             vm: {
-                                zone_state: 'installed'
+                                zone_state: 'installed',
+                                zfs_filesystem: payload.zfs_filesystem,
+                                zpool: payload.zpool,
+                                zonepath: payload.zonepath
                             }
                         };
                         var opts = {
